### ISSUE TITLE
enhancement: utilize batch in dataProcess

### ISSUE THREAD

**Issue by WenheLI** [alibaba/pipcook#411]
In our implementation right now, we apply `dataProcess` to every single sample despite what the batch size is. If we can apply the `dataProcess` at the batch level, we can definitely have a performance boost.

**Comment by yorkie** [comment]
Yes, as we have discussed before, we will deprecate the sample args and use batch instead.
